src/core/qgsvectorlayer.cpp

```diff
@@ -4335,6 +4341,27 @@ QString QgsVectorLayer::getStyleFromDatabase( const QString& styleId, QString &m
   return getStyleByIdMethod( mDataSource, styleId, msgError );
 }
 
+bool QgsVectorLayer::deleteStyleFromDatabase( const QString& styleId, QString &msgError )
+{
+  QgsProviderRegistry * pReg = QgsProviderRegistry::instance();
+  QLibrary *myLib = pReg->providerLibrary( mProviderKey );
```

Comment:
Change it to:

```
{
    QScopedPointer<QLibrary> myLib( QgsProviderRegistry::instance()->providerLibrary( mProviderKey ) );
    if ( !myLib )
    {
      msgError = QObject::tr( "Unable to load %1 provider" ).arg( mProviderKey );
      return false;
    }
    deleteStyleById_t* deleteStyleByIdMethod = reinterpret_cast< deleteStyleById_t * >( cast_to_fptr( myLib->resolve( "deleteStyleById" ) ) );
    if ( !deleteStyleByIdMethod )
    {
      msgError = QObject::tr( "Provider %1 has no %2 method" ).arg( mProviderKey, QStringLiteral( "deleteStyleById" ) );
      return false;
    }
    return deleteStyleByIdMethod( mDataSource, styleId, msgError );
}
```

When the scoped pointer goes out of scope, the associated pointer is automatically deleted.